digdag-core/src/main/java/io/digdag/core/agent/InProcessTaskCallbackApi.java

```diff
@@ -184,12 +184,13 @@ public AttemptStateFlags startSession(
                 Optional.absent());
 
         // TODO FIXME SessionMonitor monitors is not set
+        StoredSessionAttemptWithSession attempt;
         try {
-            StoredSessionAttemptWithSession attempt = exec.submitWorkflow(siteId, ar, def);
-            return attempt.getStateFlags();
+            attempt = exec.submitWorkflow(siteId, ar, def);
         }
         catch (SessionAttemptConflictException ex) {
```

Comment:
no, submitWorkflow doesn't throw `SessionAttemptConflictException` when a workflow fails. It throws `SessionAttemptConflictException` when attempt name conflicts.